backend/quatrix/quatrix.go

```diff
@@ -0,0 +1,1255 @@
+func (o *Object) ModTime(ctx context.Context) time.Time {
+	err := o.readMetaData(ctx)
+	if err != nil {
+		fs.Logf(o, "Failed to read metadata: %v", err)
+		return time.Now()
+	}
+
+	return o.modTime
+}
+
+// Storable returns a boolean showing whether this object storable
+func (o *Object) Storable() bool {
+	return true
+}
+
+// ID returns the ID of the Object if known, or "" if not
+func (o *Object) ID() string {
+	return o.id
+}
+
+// Hash returns the SHA-1 of an object. Not supported yet.
+func (o *Object) Hash(ctx context.Context, ty hash.Type) (string, error) {
+	return "", nil
+}
+
+// Remove an object
+func (o *Object) Remove(ctx context.Context) error {
+	err := o.fs.deleteObject(ctx, o.id)
+	if err != nil {
+		return err
+	}
+
+	if o.obType != "F" {
+		o.fs.dirCache.FlushDir(o.remote)
+	}
+
+	return nil
+}
+
+// Open an object for read
+func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (in io.ReadCloser, err error) {
+	if o.id == "" {
+		return nil, errors.New("can't download - no id")
+	}
+
+	linkID, err := o.fs.downloadLink(ctx, o.id)
+	if err != nil {
+		return nil, err
+	}
+
+	fs.FixRangeOption(options, o.size)
+
+	opts := rest.Opts{
+		Method:  "GET",
+		Path:    "/file/download/" + linkID,
+		Options: options,
+	}
+
+	var resp *http.Response
+
+	err = o.fs.pacer.Call(func() (bool, error) {
+		resp, err = o.fs.srv.Call(ctx, &opts)
+		return shouldRetry(ctx, resp, err)
+	})
+	if err != nil {
+		return nil, err
+	}
+	return resp.Body, err
+}
+
+func (f *Fs) downloadLink(ctx context.Context, id string) (linkID string, err error) {
+	linkParams := &api.IDList{
+		IDs: []string{id},
+	}
+	opts := rest.Opts{
+		Method: "POST",
+		Path:   "file/download-link",
+	}
+
+	var resp *http.Response
+	link := &api.DownloadLinkResponse{}
+
+	err = f.pacer.Call(func() (bool, error) {
+		resp, err = f.srv.CallJSON(ctx, &opts, linkParams, &link)
+		return shouldRetry(ctx, resp, err)
+	})
+	if err != nil {
+		return "", err
+	}
+	return link.ID, nil
+}
+
+// SetModTime sets the modification time of the local fs object
+func (o *Object) SetModTime(ctx context.Context, t time.Time) error {
+	file, err := o.fs.setMTime(ctx, o.id, t)
+	if err != nil {
+		return fmt.Errorf("set mtime: %w", err)
+	}
+
+	return o.setMetaData(file)
+}
+
+// Update the object with the contents of the io.Reader, modTime and size
+//
+// The new object may have been created if an error is returned
+func (o *Object) Update(ctx context.Context, in io.Reader, src fs.ObjectInfo, options ...fs.OpenOption) (err error) {
+	size := src.Size()
+	modTime := src.ModTime(ctx)
+	remote := o.Remote()
+
+	// Create the directory for the object if it doesn't exist
+	leaf, directoryID, err := o.fs.dirCache.FindPath(ctx, remote, true)
+	if err != nil {
+		return err
+	}
+
+	uploadSession, err := o.uploadSession(ctx, directoryID, leaf)
+	if err != nil {
+		return fmt.Errorf("object update: %w", err)
+	}
+
+	o.id = uploadSession.FileID
+
+	defer func() {
```

Comment:
I'm wondering why you need this?

Does this backend show partially uploaded files? 

If so you might be better setting the `PartialUploads` feature flag.

Reply:
Quatrix shows partially uploaded files. We can't set `PartialUploads` for now, because of behavior of Move. It does not replace a file (like a local FS), but returns the error `File exists`. I guess we will change it in a while, as separate improvement.

Reply:
Ah, OK. Leave it how it is for the moment then.

Do you have a `force` flag or something on the Move API? Rclone generally expects `Move` to overwrite existing files.

Reply:
Not yet, unfortunately. But we plan to add `overwrite` flag to API, like we have for `Copy`